modules/server/src/test/scala/higherkindness/compendium/core/ProtocolUtilsSpec.scala

```diff
@@ -44,6 +44,16 @@ object ProtocolUtilsSpec extends Specification with ScalaCheck {
         .validateProtocol(protocol)
         .unsafeRunSync must throwA[org.apache.avro.SchemaParseException]
     }
+
+    "It is possible to parse multiple protocols" >> {
+      val stream: InputStream = getClass.getResourceAsStream("/correct.avro")
+      val text: String        = scala.io.Source.fromInputStream(stream).getLines.mkString
+      val protocol: Protocol  = Protocol(text)
+
+      utils.validateProtocol(protocol).unsafeRunSync
```

Comment:
Is this needed?

Reply:
The problem occurs when making a second call with a protocol with a similar name for a record. We can refactor the code in the following way to make only one `unsafeRunSync `

```scala

      utils
        .validateProtocol(protocol)
        .flatTap(_ => utils.validateProtocol(protocol))
        .unsafeRunSync should not(throwA[org.apache.avro.SchemaParseException])
```

Reply:
+1 to that change. You can even:

`(utils.validateProtool(protocol) *> utils.validateProtocol(protocol)).unsafeRunSync`